web/views/importer.py

```diff
@@ -0,0 +1,146 @@
+    if not (has_permission and is_ajax):
+        return HttpResponseForbidden()
+    if request.method == "GET":
+        form = ImportForm(initial={"model_type": model_type})
+        return render_form_to_json(form, model_type)
+    if request.method == "POST":
+        form = ImportForm(request.POST, request.FILES)
+        if form.is_valid():
+            uploaded_file = form.cleaned_data["file"]
+            file_path = save_uploaded_file(uploaded_file, model_type)
+            output = process_uploaded_file(file_path, model_type)
+            os.remove(file_path)
+            return render_form_to_json(None, model_type, output)
+        else:
+            return render_form_to_json(form, model_type)
+
+
+def render_form_to_json(form, model_type, output=None):
+    """
+    Render the import form to a JSON response.
+
+    Args:
+        form (Form): The import form instance.
+        model_type (str): The type of the model for which data is being imported.
+        output (dict, optional): Any output data or messages to be included in the rendered HTML.
+
+    Returns:
+        JsonResponse: A JSON response containing the rendered HTML form.
+    """
+    form_html = render_to_string(
+        "importer/import_form.html",
+        {
+            "form": form,
+            "model_type": model_type,
+            "output": output,
+        },
+    )
+    return JsonResponse({"form_html": form_html})
+
+
+def save_uploaded_file(uploaded_file, model_type):
+    """
+    Save the uploaded file to a temporary location.
+
+    Args:
+        uploaded_file (UploadedFile): The uploaded file object.
+        model_type (str): The type of the model for which data is being imported.
+
+    Returns:
+        str: The path to the saved file.
+    """
+    file_path = os.path.join(settings.MEDIA_ROOT, f"temp_{model_type}.json")
+    with open(file_path, "wb+") as destination:
+        for chunk in uploaded_file.chunks():
+            destination.write(chunk)
+    return file_path
+
+
+def process_uploaded_file(file_path, model_type):
+    """
+    Process the uploaded file based on the provided model type.
+
+    This function leverages Django's call_command to run the appropriate management command for processing.
+
+    Args:
+        file_path (str): The path to the uploaded file.
+        model_type (str): The type of the model for which data is being imported.
+
+    Returns:
+        dict: A dictionary containing processing results, logs, and any errors.
+    """
+    cmd_args = [
+        f"import_{model_type}",
+        "--file",
+        file_path,
+        "--exit-on-error",
+        "--skip-on-exist",
+    ]
+
+    stdout_buffer = StringIO()
+    stderr_buffer = StringIO()
+
+    log_buffer = StringIO()
+    log_handler = logging.StreamHandler(log_buffer)
+    LOGGING_FORMAT = "%(levelname)s: %(message)s"
+    log_formatter = logging.Formatter(LOGGING_FORMAT)
+    log_handler.setFormatter(log_formatter)
+    logger = logging.getLogger()
```

Comment:
This returns the default logger right? Would it not be better to get the logger associated with the command, just in case it does not use the default one?

Reply:
i'm not sure there is one logger that i can targeted here

Reply:
Yes, the issue is that even if there is a specific loader set, there is no way to guess what the logger name will be :/